validator/node/node.go

```diff
@@ -519,6 +531,41 @@ func (c *ValidatorClient) registerRPCGatewayService(cliCtx *cli.Context) error {
 	return c.services.RegisterService(gatewaySrv)
 }
 
+func (c *ValidatorClient) registerPandoraService(cliCtx *cli.Context) error {
+	var endpoint string
+	if cliCtx.String(pandora.PandoraRpcIpcProviderFlag.Name) != "" {
+		log.WithField("ipcPath", cliCtx.String(pandora.PandoraRpcIpcProviderFlag.Name)).Info("Pandora ipc file path")
+		ipcFilePath := cliCtx.String(pandora.PandoraRpcIpcProviderFlag.Name)
+		absFilePath, err := fileutil.ExpandPath(ipcFilePath)
+		if err != nil {
```

Comment:
I think that you can simply remove those 2 `if` statements (540, 543) because in line 553 you are isomg Dial()  function which will check if client works no matter if ipc or http

Reply:
ExpandPath given a string which may be a relative path and then checks the ipc file existence in the location. If we get any error here we will not register the service. `Dial()` just specify the type of communction like http or ipc or websocket etc according to the endpoint and creates a rpc client.

Reply:
Dial() returns error on failure, your approach is more strict. I'm fine with either of them